modules/ppcp-settings/src/Data/SettingsModel.php

```diff
@@ -139,7 +139,8 @@ public function get_soft_descriptor() : string {
 	 * @param string $descriptor The soft descriptor to set.
 	 */
 	public function set_soft_descriptor( string $descriptor ) : void {
-		$this->data['soft_descriptor'] = $this->sanitizer->sanitize_text( $descriptor );
+		$trimmed_descriptor = substr( $descriptor, 0, 22 );
+		$this->data['soft_descriptor'] = $this->sanitizer->sanitize_text( $trimmed_descriptor );
```

Comment:
Practically, this is working well, but generally, we should:

- First: Sanitize the value
- Second: Modify/use the sanitized value

i.e.
```php
// First sanitize, to remove invalid characters/text (like <script> tags)
$sanitized_descriptor = $this->sanitizer->sanitize_text( ... );

// Second, trim and store the sanitized value
$this->data['soft_descriptor'] = substr( $sanitized_descriptor, 0, 22 );
```